executor/executor.go

```diff
@@ -175,6 +175,17 @@ type Executor interface {
 	newFirstChunk() *chunk.Chunk
 }
 
+// Next is a wrapper function on e.Next(), it handles some common codes.
+func Next(ctx context.Context, e Executor, req *chunk.RecordBatch) error {
+	select {
+	case <-ctx.Done():
+		req.Reset()
+		return errors.Trace(ctx.Err())
```

Comment:
no need to trace

Reply:
This is not a defined error in TiDB, I'm afraid without trace there won't be an error stack.
Or, should we define an error for it?